tests/linearizability/model.go

```diff
@@ -151,3 +164,15 @@ func stepPut(state EtcdState, request etcdRequest, response etcdResponse) (bool,
 	state.LastRevision = response.revision
 	return true, state
 }
+
+func stepDelete(state EtcdState, request etcdRequest, response etcdResponse) (bool, EtcdState) {
+	if response.err != nil {
+		state.FailedWrites[request.putData] = struct{}{}
+		return true, state
+	}
+	if response.deleted == 1 && state.LastRevision >= response.revision {
```

Comment:
```suggestion
    deleteSucceded := response.deleted != 0
    keySet := state.Value != ""
	if (deleteSucceded != keySet) || state.LastRevision >= response.revision {
```

Delete request is invalid if either:
* response says there was a key deleted (`reponse.deleted != 0`) but the key was not present (`state.Value == ""`)
* response says there was a key was not deleted (`reponse.deleted == 0`) but the key was present (`state.Value != ""`)
* revision has not increased  `state.LastRevision >= response.revision`

all cases:

|state.Value | reponse.deleted | correct? |  explanation  |
|-----------------|-------------------------|--------------|-------------------|
| "a"                | 1                           | true         | if key was set, delete should return that it was deleted.
| "a"                | 0                           | false        | if key was set, delete cannot return that it didn't delete any keys
| ""                  | 1                           | false        | if key was not set, delete cannot return that it deleted it.
| ""                  | 0                           | true         | if key was not set, delete should report that it didn't delete anything